gammapy/background/models.py

```diff
@@ -71,3 +81,775 @@ def evaluate(self, x, y):
+        bin_pos = np.array([bin_pos_x, bin_pos_y])
+        bin_edges = Angle([detx_edges_low[bin_pos[0]], detx_edges_high[bin_pos[0]],
+                           dety_edges_low[bin_pos[1]], dety_edges_high[bin_pos[1]]])
+
+        return bin_pos, bin_edges
+
+    def find_energy_bin(self, energy):
+        """Find the bin that contains the specified energy value.
+
+        TODO: implement test as suggested in:
+            https://github.com/gammapy/gammapy/pull/292#discussion_r33843508
+
+        Parameters
+        ----------
+        energy : `~astropy.units.Quantity`
+            energy to search for
+
+        Returns
+        -------
+        bin_pos : `~int`
+            index of the energy bin containing the specified energy
+        bin_edges : `~astropy.units.Quantity`
+            energy bin edges [E_min, E_max)
+        """
+        # check shape of energy: only 1 value is accepted
+        nvalues = len(energy.flatten())
+        if nvalues != 1:
+            raise IndexError("Expected exactly 1 value for energy, got {}."
+                             .format(nvalues))
+
+        # check that the specified energy is within the boundaries of the model
+        energy_extent = self.spectrum_extent
+        if not (energy_extent[0] <= energy) and (energy < energy_extent[1]):
+            ss_error = "Specified energy {}".format(energy)
+            ss_error += " is outside the boundaries {}.".format(energy_extent)
+            raise ValueError(ss_error)
+
+        energy_edges_low = self.energy_bins[:-1]
+        energy_edges_high = self.energy_bins[1:]
+        bin_pos = np.searchsorted(energy_edges_high, energy)
+        bin_edges = Quantity([energy_edges_low[bin_pos], energy_edges_high[bin_pos]])
+
+        return bin_pos, bin_edges
+
+    def plot_image(self, energy, ax=None, save=False):
+        """Plot image for the energy bin containing the specified energy.
+
+        Parameters
+        ----------
+        energy : `~astropy.units.Quantity`
+            energy of bin to plot the bg model
+        ax : `~matplotlib.axes.Axes`, optional
+            axes of the figure
+        save : `~bool`, optional
+            flag to turn on/off the saving of the figure in a png file
+
+        Returns
+        -------
+        fig : `~matplotlib.figure.Figure`
+            figure with image of bin of the bg model for the
+            selected energy value
+        ax : `~matplotlib.axes.Axes`
+            axes of the figure
+        image : `~matplotlib.image.AxesImage`
+            image of the figure, optional
+        """
+        import matplotlib.pyplot as plt
+        from matplotlib.colors import LogNorm
+        from mpl_toolkits.axes_grid1 import make_axes_locatable
+
+        energy = energy.flatten() # flatten
+        # check shape of energy: only 1 value is accepted
+        nvalues = len(energy)
+        if nvalues != 1:
+            print("Expected exactly 1 value for energy, got {}.".format(nvalues))
```

Comment:
You should never print from library code.
It makes it impossible for users to turn the print statements off.
MINUIT e.g. and other legacy packages are doing this wrong and are not nice to use in large batch processing.

You can just put that message into the `IndexError(message)` call or remove the print statement altogether.